activerecord/lib/active_record/associations/preloader/batch.rb

```diff
@@ -16,10 +16,7 @@ def call
 
             loaders.each { |loader| loader.associate_records_from_unscoped(@available_records[loader.klass]) }
 
-            already_loaded = loaders.select(&:data_available?)
```

Comment:
We can't run the `already_loaded` loaders first if we want `test_preload_grouped_queries_with_already_loaded_records` to pass because we lose the grouping.